core/node/node_framework/src/implementations/resources/pools.rs

```diff
@@ -95,10 +95,14 @@ impl ReplicaPoolResource {
     }
 
     pub async fn get_singleton(&self) -> anyhow::Result<ConnectionPool> {
+        self.get_custom(1).await
+    }
+
+    pub async fn get_custom(&self, size: u32) -> anyhow::Result<ConnectionPool> {
```

Comment:
I don't quite understand how this method works. Wouldn't it always build a singleton pool, regardless of the `size` supplied? It seems that it only influences logs.

Reply:
Oh, right. What was I even thinking? 🤦